### ISSUE TITLE
Revamp invite links

### ISSUE THREAD

**Issue by SoniEx2** [tootsuite/mastodon#13821]
### Pitch

When you give someone an invite link, and they open it, they get a sign-up form with little to no information. This is in contrast to non-invite links in e.g. open instances. It'd be hugely beneficial if the page introduced the instance/network to the user.

### Motivation

I'd like to share invite links with friends and let the instance explain itself. While I don't mind having to explain the network, having to explain the network *and* the instance feels like it'll just keep ppl from joining, somehow? The idea here is that I think I should get ppl interested on the network, and then they open the invite link and learn more about the specific instance I picked for them. They can then choose to pick another instance, give up entirely, or go with it.

In any case, being met with a sign up page with no further information is a good way to scare ppl off tbh. :p

**Comment by ThibG** [comment]
I wonder if having almost exactly the same layout as /about would be an issue, and if not, what would be needed to make it clearer that you've been just invited to the instance, what's expected of you is to register, and that you may be unable to do so by doing something else than clicking the invitation link.

**Comment by revk** [comment]
Can I also make a further suggestion on this. At present it seems I can have an open registration form, or a registration form that then needs my approval, or no registration form. If I make an invite link though, it always works with no approval.

For a start, the form that needs approval would be good if it said you will need approval. Or maybe at least a configureable paragraph to show.

But also, could we have a mode where the invite link also needs approval. An option on the link when generated maybe.

Basically, on an otherwise closed server, I invited a few friends with one link to them (so unlimited usage), and one of them, smart arse, decided to also create realDonaldTrump and ElonMusk. Would be nice if even the invite links had an approval mode as well.

**Comment by SuperSandro2000** [comment]
@revk generate an invite link with only one use.

**Comment by revk** [comment]
I agree that is an option, but would be nice to have multiple use but "approval mode", that is all I am saying.

**Comment by SoniEx2** [comment]
is there any way we can help with this?
